### ISSUE TITLE
"triggered runs" dataset counter doesn't update until *next* run and never goes above 1

### ISSUE THREAD

**Issue by MatrixManAtYrService** [apache/airflow#26256]
### Apache Airflow version

2.4.0b1

### What happened

I have [this test dag](https://gist.github.com/MatrixManAtYrService/2cf0ebbd85faa2aac682d9c441796c58) which I created to report [this issue](https://github.com/apache/airflow/issues/25210).  The idea is that if you unpause "sink" and all of the "sources" then the sources will wait until the clock is like \*:\*:00 and they'll terminate at the same time.  

Since each source triggers the sink with a dataset called "counter", the "sink" dag will run just once, and it will have output like:  `INFO - [(16, 1)]`, that's 16 sources and 1 sink that ran.

At this point, you can look at the dataset history for "counter" and you'll see this:

<img width="524" alt="Screen Shot 2022-09-08 at 6 07 44 PM" src="https://user-images.githubusercontent.com/5834582/189248999-d31141a4-2d0b-4ec2-9ea5-c4c3536b3a28.png">

So we've got a timestamp, but the "triggered runs" count is empty.  That's weird.  One run was triggered (and it finished by the time the screenshot was taken), so why doesn't it say `1`?

So I redeploy and try it again, except this time I wait several seconds between each "unpause" click, the idea being that maybe some of them fire at 07:16:00 and the others fire at 07:17:00.  I end up with this:

<img width="699" alt="Screen Shot 2022-09-08 at 6 19 12 PM" src="https://user-images.githubusercontent.com/5834582/189252116-69067189-751d-40e7-89c5-8d1da1720237.png">

So fifteen of them finished at once and caused the dataset to update, and then just one straggler (number 9)  is waiting for an additional minute.  I wait for the straggler to complete and go back to the dataset view:

<img width="496" alt="Screen Shot 2022-09-08 at 6 20 41 PM" src="https://user-images.githubusercontent.com/5834582/189253874-87bb3eb3-2237-42a1-bc3f-9fc210419f1a.png">

Now it's the straggler that is blank, but the rest of them are populated.  Continuing to manually run these, I find that whichever one I have run most recently is blank, and all of the others are 1, even if this is the second or third time I've run them

### What you think should happen instead

- The triggered runs counter should increment beyond 1
- It should increment immediately after the dag was triggered, not wait until after the *next* dag gets triggered.

### How to reproduce

See dags in in this gist: https://gist.github.com/MatrixManAtYrService/2cf0ebbd85faa2aac682d9c441796c58

1. unpause "sink"
2. unpause half of sources
3. wait one minute
4. unpause the other half of the sources
5. wait for "sink" to run a second time
6. view the dataset history for "counter"
7. ask why only half of the counts are populated
8. manually trigger some sources, wait for them to trigger sink
9. view the dataset history again
10. ask why none of them show more than 1 dagrun triggered

### Operating System

Kubernetes in Docker, deployed via helm

### Versions of Apache Airflow Providers

n/a

### Deployment

Other 3rd-party Helm chart

### Deployment details

see "deploy.sh" in the gist: 

https://gist.github.com/MatrixManAtYrService/2cf0ebbd85faa2aac682d9c441796c58

It's just a fresh install into a k8s cluster

### Anything else

n/a

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by jedcunningham** [comment]
So there are a couple of things going on here.

The first is that the "triggered dagruns" column is empty sometimes. This is being fixed in #26256.

Next is that you expect greater than 1 with the DAGs you are testing with. You have many updates to a single dataset, and a single downstream consumer. With this setup, by definition each dataset event can only trigger a single downstream run - even though there are many events, each specific event can only trigger the one downstream DAG once. In contrast, the follow setup where there are multiple consumers of the dataset will properly show more than 1 triggered dagrun per event.

```python
from datetime import datetime

from airflow import DAG, Dataset
from airflow.operators.empty import EmptyOperator

dataset = Dataset("hello")

with DAG(
    dag_id="dataset_many_upstream",
    start_date=datetime(2022, 8, 1),
    schedule="@daily",
):
    EmptyOperator(task_id="upstream_task", outlets=[dataset])

with DAG(
    dag_id="dataset_many_downstream1",
    start_date=datetime(2022, 8, 1),
    schedule=[dataset],
):
    EmptyOperator(task_id="downstream")

with DAG(
    dag_id="dataset_many_downstream2",
    start_date=datetime(2022, 8, 1),
    schedule=[dataset],
):
    EmptyOperator(task_id="downstream")
```

![Screen Shot 2022-09-09 at 10 29 21 AM](https://user-images.githubusercontent.com/66968678/189401731-b9c69a8d-789a-4d1f-8717-2f70949848c0.png)

**Comment by MatrixManAtYrService** [comment]
Oh I see, so it's not: 
 > how many runs has this dataset triggered ever?
 
But instead:
> the last time this dataset was updated, how many runs did it trigger?
